torch/_subclasses/meta_utils.py

```diff
@@ -964,7 +964,13 @@ def symint_visitor_fn(s):
                 # assumption of it being simplified out will fail and it may be guarded on,
                 # which will hard error.
                 sym_source = EphemeralSource("symint_visitor_fn")
-                symbol = shape_env.create_symbol(s, sym_source)
+
+                positive = True
+                if isinstance(s, int) and s < 0:
+                    # e.g. if you .view(-1, ...)
+                    positive = False
+
+                symbol = shape_env.create_symbol(s, sym_source, positive=positive)
```

Comment:
cc @ezyang in case you have an opinion on whether `positive=True` or `positive=None` is better to use here when we don't have `s < 0`